soc/arm/nuvoton_npcx/common/soc_dt.h

```diff
@@ -705,4 +705,16 @@
 #define NPCX_DT_PSL_OUT_PIN(inst) DT_PROP(DT_INST(inst, nuvoton_npcx_psl_out), \
 							pin)
 
+/**
+ * @brief Check if the host interface type is automatically configured by
+ * booter.
+ *
+ * @param @param inst number for devices with compatible
+ *               'nuvoton_npcx_booter_variant'.
+ * @return TRUE - if the host interface is configured by booter,
+ *         FALSE - otherwise.
+ */
+#define NPCX_BOOTER_IS_HIF_TYPE_SET() \
+	DT_PROP(DT_INST(0, nuvoton_npcx_booter_variant), hif_type_auto)
```

Comment:
Interesting use of DT.

I think `DT_PROP(DT_PATH(soc, booter_variant), hif_type_auto)` would be a bit more robust if you're always going to have this node at `/soc/booter-variant`. Sometimes people get confused and try to add nodes of different compatibles. If that happens here, you wouldn't be sure which node you're reading from if you use instance numbers.

Just something to consider; feel free to disregard.